pykern/pkunit.py

```diff
@@ -278,6 +283,29 @@ def file_eq(expect_path, *args, **kwargs):
     _FileEq(expect_path, *args, **kwargs)
 
 
+def unbound_localhost_tcp_port(start, stop):
+    """Looks for AF_INET SOCK_STREAM port for which bind fails
```

Comment:
Shouldn't it be "...for which bind __succeeds__" or maybe better "...port that we are able to bind to"?

Reply:
If the bind fails, the port is "unbound" and therefore free to use as a server.

Reply:
I guess I don't understand what "bind fails" means? Fails implies some sort of error to me. From the code below it looks to me that if something is already bound to the ip:port then bind "fails". 

```bash
~$ lsof -i -P | grep 12345
~$ python <<EOF
> import socket

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind(("127.0.0.1", 12345))
    print("after bind")
EOF

> > > > > after bind
~$ python -m http.server 12345 &
[1] 26826
~$ Serving HTTP on 0.0.0.0 port 12345 (http://0.0.0.0:12345/) ...

~$ lsof -i -P | grep 12345
python  26826 vagrant    3u  IPv4 1547964      0t0  TCP *:12345 (LISTEN)
~$ python <<EOF
> import socket

with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
    s.bind(("127.0.0.1", 12345))
    print("after bind")
EOF

> > > > > Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
OSError: [Errno 98] Address already in use
```

Reply:
b3b10bb646. boolean confusion, sorry